### DISCUSSION TITLE
Read data from json file

### DISCUSSION THREAD

**Discussion by FadelOj** [cypress-io/cypress#18812] [Component Testing]
Hello everyone, 

I wanted to make a test case that will read the properties of the user (username, name, password, email ..... ) from the users.json file and fill them in the corresponsive field in the website that I am testing.

`      
       cy.get('#IndividualDetailViewModel_EnglishFullName').invoke('val', 'englishName')
       cy.get('#IndividualDetailViewModel_PhoneNumber').invoke('val', '000000000')
       cy.get('#IndividualDetailViewModel_IdNumber').invoke('val', '00000000000')
       cy.get('#IndividualDetailViewModel_UserName').invoke('val', 'XXXXXXXXXX')
       cy.get('#IndividualDetailViewModel_Email').invoke('val', 'XXXXXX@XXXXX.com')
       cy.get('#IndividualDetailViewModel_Password').invoke('val', 'P@ssw0rd')
       cy.get('#IndividualDetailViewModel_ConfirmPassword').invoke('val', 'P@ssw0rd')
`
   I have read the documentation already, so if someone can help me with the code directly, thanks in advance.

   [users.txt](https://github.com/cypress-io/cypress/files/7492275/users.txt)

**Comment by bahmutov** [comment]
You can use cy.readFile or cy.fixture commands or directly import the json file from the spec

**Reply by FadelOj** [reply]
thank you for the tips, but I was look for the straight forward answer because i have stuck with this issue for 3 days

**Reply by FadelOj** [reply]
I have tried cy.fixture commands, but they didn't mention how can retrieve the values and use them as input for the testscript, thats what i am trying to do, getting the element from json then use it as an input value

**Comment by bahmutov** [comment]
Ok. In that case search for json and fixture keywords at https://cypress.tips/search and read the search results and watch the videos
